### PR TITLE
fix: Empty Embed Error

### PR DESCRIPTION
**Pull request by devnote-dev**
Due to a change with Discord Embeds, empty embeds can no longer be created and an empty description field is no longer accepted. This means that currently the `simpleembed` and `customembed` commands throw the following error when no description is provided: `HTTP 400 Bad Request, {"embed": ["description"]}`. This PR fixes this, as well as three additional bugs with the commands:
- URL field not included if no Title is provided
- Author field not included if no name is provided
- Footer field not included if no text is provided

I'm no expert with Golang so additional reviews/testing is requested please, thanks!

### CODE REVIEW

**Comment by jonas747** [comment]
sending "\u200b" is a bit too hacky IMO, we should instead error out with a more helpfull message

**Comment by devnote-dev** [comment]
ZWS for description has been removed 👍

**Comment by jonas747** [comment]
Embeds are valid without a description if you provide a title for example

**Comment by devnote-dev** [comment]
Ah my bad, latest commits should now check for required/non-required fields now before sending

File: stdcommands/simpleembed/simpleembed.go [hunk 1]
```diff
@@ -30,14 +30,28 @@ var Command = &commands.YAGCommand{
...
+		&dcmd.ArgDef{Switch: "footer", Help: "Text content for the footer", Type: dcmd.String, Default: "\u200b"},
 		&dcmd.ArgDef{Switch: "footericon", Help: "Url to a icon for the 'footer' field", Type: dcmd.String, Default: ""},
 	},
 	RunFunc: func(data *dcmd.Data) (interface{}, error) {
+		if data.Switch("channel").Str() != "" || 
+			data.Switch("content").Str() != "" || 
+			data.Switch("color").Str() != "" || 
+			data.Switch("url").Str() != "" || 
+			data.Switch("authorurl").Str() != "" {
+    			if data.Switch("title").Str() == "" &&
```
Comment (jonas747) [inline stdcommands/simpleembed/simpleembed.go:46]: I don't quite understand why the parent if statement needs to exist

File: stdcommands/simpleembed/simpleembed.go [hunk 2]
```diff
@@ -30,14 +30,28 @@ var Command = &commands.YAGCommand{
 		&dcmd.ArgDef{Switch: "thumbnail", Help: "Url to a thumbnail", Type: dcmd.String, Default: ""},
 		&dcmd.ArgDef{Switch: "image", Help: "Url to an image", Type: dcmd.String, Default: ""},
 
-		&dcmd.ArgDef{Switch: "author", Help: "The text in the 'author' field", Type: dcmd.String, Default: ""},
+		&dcmd.ArgDef{Switch: "author", Help: "The text in the 'author' field", Type: dcmd.String, Default: "\u200b"},
```
Comment (jonas747) [inline stdcommands/simpleembed/simpleembed.go:33]: u200b is still present in the defaults

**Review by jonas747 — CHANGES_REQUESTED** [review]
Im a bit puzzled by the neted if statements

**Comment by ashishjh-bst** [comment]
review comments from jonas are still not resolved, plus the branch needs to be rebased.

**Comment by devnote-dev** [comment]
I will reopen this when I have time and a working testing environment (or someone else can fix this).

**Pull request closed without merging**